app/src/main/java/com/example/noteandtodo/MyContentProvider.java

```diff
@@ -0,0 +1,266 @@
+package com.example.noteandtodo;
+
+import android.content.ContentProvider;
+import android.content.ContentResolver;
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.content.UriMatcher;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteQueryBuilder;
+import android.net.Uri;
+import android.os.Bundle;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * ContentProvider for note and todo
+ *
+ * Created on 2016/04/12.
+ */
+public class MyContentProvider extends ContentProvider{
```

Comment:
Also I noticed that `todo` checkbox state gets lost when converting to and from a `note`...